src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -737,38 +732,6 @@ public ProjectState WithAnalyzerReferences(IEnumerable<AnalyzerReference> analyz
         return With(projectInfo: ProjectInfo.WithAnalyzerReferences(analyzerReferences).WithVersion(Version.GetNewerVersion()));
     }
 
-    [MemberNotNull(nameof(_lazySourceGenerators))]
-    private void EnsureSourceGeneratorsInitialized()
-    {
-        if (_lazySourceGenerators == null)
-        {
-            var builder = ImmutableDictionary.CreateBuilder<ISourceGenerator, AnalyzerReference>();
-
-            foreach (var analyzerReference in AnalyzerReferences)
-            {
-                foreach (var generator in analyzerReference.GetGenerators(Language))
-                    builder.Add(generator, analyzerReference);
-            }
-
-            Interlocked.CompareExchange(ref _lazySourceGenerators, builder.ToImmutable(), comparand: null);
-        }
-    }
-
-    public IEnumerable<ISourceGenerator> SourceGenerators
```

Comment:
this was too broad a surface area.  we def did not want to expose inside the workspace that you could get to ISourceGenerator instances.